core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAO.scala

```diff
@@ -187,16 +187,20 @@ class HttpWorkspaceManagerDAO(apiClientProvider: WorkspaceManagerApiClientProvid
                                                description: Option[DataReferenceDescriptionField],
                                                instanceName: String,
                                                cloningInstructions: CloningInstructionsEnum,
+                                               properties: Option[Map[String, String]],
                                                ctx: RawlsRequestContext
   ): DataRepoSnapshotResource = {
     val snapshot = new DataRepoSnapshotAttributes().instanceName(instanceName).snapshot(snapshotId.toString)
     val commonFields =
       new ReferenceResourceCommonFields()
         .name(name.value)
-        // Note: that we're ignoring the passed in `cloningInstructions` is a known issue, addressed as part of
-        // https://github.com/broadinstitute/rawls/pull/2081
-        .cloningInstructions(CloningInstructionsEnum.NOTHING)
+        .cloningInstructions(cloningInstructions)
     description.map(d => commonFields.description(d.value))
+
+    val requestProperties = new Properties
+    properties.foreach(p => p.foreach(kv => requestProperties.add(new Property().key(kv._1).value(kv._2))))
```

Comment:
FWIW you could also do this to avoid the `._1` and `._2` with the tuple. Totally up to you on this since what you have here is perfectly clear IMO.
```suggestion
    properties.foreach(p =>
      p.foreach { case (key, value) => requestProperties.add(new Property().key(key).value(value)) }
    )```